python/kserve/kserve/storage/storage.py

```diff
@@ -158,6 +160,24 @@ def _update_with_storage_spec():
                     f.write(value)
                     f.flush()
 
+        if storage_secret_json.get("type", "") == "gs":
+            if storage_secret_json.get("base64_service_account_key_file", "") != "":
+                try:
+                    base64_service_account_key_file = storage_secret_json.get(
+                        "base64_service_account_key_file", ""
+                    )
+                    service_account_str = base64.b64decode(
+                        base64_service_account_key_file
+                    ).decode("utf-8")
+                    service_account_dict = json.loads(service_account_str)
+                    os.environ["GOOGLE_SERVICE_ACCOUNT"] = json.dumps(
```

Comment:
Seems like we should use the environment variable `GOOGLE_APPLICATION_CREDENTIALS` to point to the service account key file.

https://cloud.google.com/docs/authentication/application-default-credentials

Reply:
We can mount the secret in the container

Reply:
Hi @yuzisun, can I clarify if I am understanding correctly:

In `CreateStorageSpecSecretEnvs`, if the storage spec secret type is gcs, call `mountSecretCredential` which calls `gcs.BuildSecretVolume`. The function will need to be modified to extract the specific gcs object from the storage spec secret , and build the volume based off its `base64_service_account_key_file` value.

Reply:
@tjandy98 sorry for the late reply, you are right. We build the volume point to the mounted base64 sa key file and env `GOOGLE_APPLICATION_CREDENTIALS` will be taking effect when calling the GCS API.